### ISSUE TITLE
findAndCountAll with Sequelize.or() throwing "unhandled TypeError: Object [object Object] has no method 'push'"

### ISSUE THREAD

**Issue by MtDalPizzol** [sequelize/sequelize#3253]
Hello everyone! I'm trying to filter some records on my DB using Sequelize.or(). It generates the correct query, but it throws an error before I can even log the results.

### The code
```javascript
var query = {
    include: [{
        model: model.Email,
        attributes: ['email'],
        where: {
            main: true
        }
    }, {
        model: model.Phone,
        attributes: ['number'],
        where: {
            main: true
        }
    }],
    offset: req.pagination.offset,
    limit: req.pagination.rowsPerPage
};

// Everything works fine if I remove the following condition
if (req.query.name) {
    query.where = model.Sequelize.or(
                    { first_name : { like: '%' + req.query.name + '%' } },
                    { last_name : { like: '%' + req.query.name + '%' } }
                  );
}

model.User.findAndCountAll(query).then(function(users) {
    console.log(users);
});
```

### The generated query
```
SELECT count("User"."id") AS "count" FROM "user" AS "User" INNER JOIN "email" AS "Emails" ON "User"."id" = "Emails"."user_id" AND "Emails"."main"=true INNER JOIN "phone" AS "Phones" ON "User"."id" = "Phones"."user_id" AND "Phones"."main"=true WHERE ("User"."first_name" LIKE '%john%' OR "User"."last_name" LIKE '%doe%');
```

### The error
```
Possibly unhandled TypeError: Object [object Object] has no method 'push'
    at generateJoinQueries (/home/ubuntu/workspace/app/node_modules/sequelize/lib/dialects/abstract/query-generator.js:1127:36)
    at Object.<anonymous> (/home/ubuntu/workspace/app/node_modules/sequelize/lib/dialects/abstract/query-generator.js:1165:29)
    at Array.forEach (native)
    at Object.module.exports.QueryGenerator.selectQuery (/home/ubuntu/workspace/app/node_modules/sequelize/lib/dialects/abstract/query-generator.js:1164:25)
    at module.exports.QueryInterface.select (/home/ubuntu/workspace/app/node_modules/sequelize/lib/query-interface.js:679:35)
    at null.<anonymous> (/home/ubuntu/workspace/app/node_modules/sequelize/lib/model.js:728:34)
```

**Comment by mickhansen** [comment]
Looks to be a regression bug, try this for a fix:

```js
query.where = {
  $or: {
    first_name: {},
    last_name: {}
  }
}
```

**Comment by mickhansen** [comment]
Or hmm, actually not sure that will work.

**Comment by MtDalPizzol** [comment]
I think it can be a bug with the findAndCountAll() function.

This function is generating only the first query (the one which does the count). Rows aren't even created.

I have already tried this alternative, @mickhansen but unfortunally I get an error saying that the $or column does not exist...

**Comment by BridgeAR** [comment]
@mickhansen This is no regression and probably fixed.
@MtDalPizzol It seemed like you use a very old version. Please try again with 2.1.3 (or 3.0.1 after the release).

**Comment by BridgeAR** [comment]
Ping @MtDalPizzol

**Comment by MtDalPizzol** [comment]
Sorry for the delay, @BridgeAR . I'm using 2.0.3. I actually stopped trying to make magic with the ORM stuff and I decided to work exclusively with raw queries... I was losing a lot of time trying to write beautiful codes with the ORM utilities...

**Comment by BridgeAR** [comment]
@MtDalPizzol That's too bad. I'd say sequelize is getting quite stable. I'd also recommend to switch to 3.x.x.
Besides that: would you be so kind and test if your old problem still exists? That would be a lot of help :)

**Comment by MtDalPizzol** [comment]
Ok. @BridgeAR ! To be honest, I didn't noticed, until you mention it, that so many versions have been released after 2.0.3. I'll give it a try this week and I'll let you guys know the results as soon as possible.
